### PR TITLE
Split repository into a library and CLI

### PR DESCRIPTION
**Pull request by BlackHoleFox**
This makes the big split of breaking up the repository into a workspace and making `libsignify` separate from the `signify` CLI tool. I took the approach of `signify` and `libsignify` for:
1. Most users probably only want the CLI and its functionality. `cargo install signify` is much nicer then `cargo install signify-cli`.
2. `clap` and `rpassword` aren't good dependencies to have in a regular library, which would be required if the CLI just became `src/cli/main.rs` in the `signify` crate.

I thought `libsignify` was the best name out of `signify-lib` and others. Please feel free to recommend anything better since it isn't perfect.

I attempted (and failed a few times) to design the library's API to be mostly dead-simple. It gets to the point of signify: it can sign and verify files and messages. There aren't any internal details exposed about the "scheme" itself, like comment headers, lengths, etc. Another goal of the design was friendliness towards eventually making it `no_std` compatible (and maybe `no_alloc`) for low-level consumers. For example: maybe an OS kernel or software updater. Slices are preferred most of the time and external reliance on `std::io::{Read, Write}` is removed. I think its use internally can be cleaned up later. However, even given this, the APIs around deserialization still provide hints to allow for consumers with `std` to make use of it in regards to memory efficiency.

@badboy if you get a chance, please take a look and see if this aligns with your ideals too.

Depends on #36 
Part 2/2 of the bigger lib/CLI split.

I plan on following up with more docs, README updates, and general repo structure cleanup after this. I didn't do it here in the interest of keeping it easier to review. I dropped the crate's `include = [...]` sections until this is worked out for simplicity too.

### CODE REVIEW

**Comment by BlackHoleFox** [comment]
I also changed the Makefile to require bash to run the test scripts (no one only has `sh` these days, right?). Execute bits on files in Git are shockingly fragile when Windows is involved.

**Comment by badboy** [comment]
WUHU!
I'm putting this on my list, hopefully for the weekend.
Thanks for the thorough explanation, overall I think that's a very good approach.

File: libsignify/src/consts.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+//! Constants and type defintions from the `signify` design.
```
Comment (badboy) [inline libsignify/src/consts.rs]:
```suggestion
//! Constants and type definitions from the `signify` design.
```

File: libsignify/src/errors.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Error::Io(e) => Display::fmt(e, f),
+            Error::InvalidFormat(e) => Display::fmt(e, f),
+            Error::UnsupportedAlgorithm => f.write_str("encountered unsupported key algorithm"),
+            Error::MismatchedKey { expected, found } => write!(f,
+                "failed to verify signature: the wrong key was used. Expected #{:?}, but found {:?}",
```
Comment (badboy) [inline libsignify/src/errors.rs]:
```suggestion
                "failed to verify signature: the wrong key was used. Expected {:?}, but found {:?}",
```

File: signify/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "signify"
+version = "0.4.1"
+authors = ["Jan-Erik Rediger <janerik@fnordig.de>"]
```
Comment (badboy) [inline signify/Cargo.toml]: we can drop this line, alternatively add yourself here

File: signify/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+license = "MIT"
+
+homepage = "https://github.com/badboy/signify-rs"
+repository = "https://github.com/badboy/signify-rs"
+
+[dependencies]
+clap = { version = "3.0.0-beta.5", default-features = false, features = ["cargo", "derive", "std"] }
+rand_core = { version = "0.5", features = ["getrandom"] }
+rpassword = "5"
+libsignify = { path = "../libsignify" }
```
Comment (badboy) [inline signify/Cargo.toml:20]: this will require a version before publishing

**Review by badboy — APPROVED** [review]
looks great! minimal nits, but nothing that really blocks this.

File: libsignify/src/consts.rs (hunk 1 above)
Comment (BlackHoleFox) [reply libsignify/src/consts.rs]: Ouch, thanks. I flipped a spellchecker on and took a look through the project and didn't see any other typos.

File: libsignify/src/errors.rs (hunk 2 above)
Comment (BlackHoleFox) [reply libsignify/src/errors.rs]: Cleaned this up too.

File: signify/Cargo.toml (hunk 4 above)
Comment (BlackHoleFox) [reply signify/Cargo.toml:20]: Indeed, yep. I still plan on doing another PR or two to clean and improve things more before going for a release. I think in the last one of those I'll switch this to a version.

File: signify/Cargo.toml (hunk 3 above)
Comment (BlackHoleFox) [reply signify/Cargo.toml]: Added myself here for now. Happy to drop the line or switch it to "Signify Contributors" if the crate picks up more users and contributors.

**Comment by BlackHoleFox** [comment]
@badboy Many thanks for the review, I'll try to have the next set of PRs up soon.

**Pull request merged**